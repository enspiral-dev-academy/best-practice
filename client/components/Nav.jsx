import React, { useEffect } from "react";
import { Link } from "react-router-dom";

import { wrappedWith, authentication, userContext } from "../wrappers";

const groupStyle = {
  float: "right",
};

const linkStyle = {
  marginRight: "30px",
};

export function Nav({ authenticated, setUserIfLoggedIn, logOut }) {
  useEffect(() => {
    setUserIfLoggedIn();
  }, []);
  return (
    <>
      <div style={groupStyle}>
        <Link to="/" style={linkStyle}>
          Home
        </Link>
        {authenticated() ? (
          <Link to="#" data-testid="logoff" style={linkStyle} onClick={logOut}>
            Log off
          </Link>
        ) : (
          <>
            <Link to="/register" data-testid="register" style={linkStyle}>
              Register
            </Link>
            <Link to="/signin" data-testid="signin" style={linkStyle}>
              Sign in
            </Link>
          </>
        )}
      </div>
    </>
  );
}

export default wrappedWith(authentication, userContext)(Nav);
