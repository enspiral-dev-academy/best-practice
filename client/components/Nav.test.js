import React from "react";
import "@testing-library/jest-dom";
import { screen, fireEvent } from "@testing-library/react";
import { renderWithRouter } from "../testing/utils";

import { Nav } from "./Nav";

test("renders correctly if authenticated", () => {
  const { asFragment } = renderWithRouter(
    <Nav authenticated={() => true} setUserIfLoggedIn={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders correctly if not authenticated", () => {
  const { asFragment } = renderWithRouter(
    <Nav authenticated={() => false} setUserIfLoggedIn={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("calls setUserIfLoggedIn on mount", () => {
  const setUserIfLoggedIn = jest.fn();
  renderWithRouter(
    <Nav authenticated={() => {}} setUserIfLoggedIn={setUserIfLoggedIn} />
  );
  expect(setUserIfLoggedIn).toHaveBeenCalled();
});

test("calls logOut on log off link click", async () => {
  const logOut = jest.fn();
  renderWithRouter(
    <Nav
      authenticated={() => true}
      setUserIfLoggedIn={() => {}}
      logOut={logOut}
    />
  );
  const logOffLink = await screen.getByRole("link", { name: "Log off" });
  fireEvent.click(logOffLink);

  expect(logOut).toHaveBeenCalled();
});
