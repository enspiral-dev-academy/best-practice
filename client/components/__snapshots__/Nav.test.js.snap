// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if authenticated 1`] = `
<DocumentFragment>
  <div
    style="float: right;"
  >
    <a
      href="/"
      style="margin-right: 30px;"
    >
      Home
    </a>
    <a
      data-testid="logoff"
      href="/"
      style="margin-right: 30px;"
    >
      Log off
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders correctly if not authenticated 1`] = `
<DocumentFragment>
  <div
    style="float: right;"
  >
    <a
      href="/"
      style="margin-right: 30px;"
    >
      Home
    </a>
    <a
      data-testid="register"
      href="/register"
      style="margin-right: 30px;"
    >
      Register
    </a>
    <a
      data-testid="signin"
      href="/signin"
      style="margin-right: 30px;"
    >
      Sign in
    </a>
  </div>
</DocumentFragment>
`;
